hcp/common.go

```diff
@@ -0,0 +1,145 @@
+package hcp
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/hcp-sdk-go/clients/cloud-packer-service/stable/2021-04-30/models"
+	sdkpacker "github.com/hashicorp/packer-plugin-sdk/packer"
+	"github.com/hashicorp/packer/internal/registry"
+	"github.com/mitchellh/mapstructure"
+
+	registryimage "github.com/hashicorp/packer-plugin-sdk/packer/registry/image"
+)
+
+func populateIteration(
+	ctx context.Context,
+	handler HCPHandler,
+) error {
+	err := handler.Bucket().Initialize(ctx)
```

Comment:
We could ensure that, but since this is a private function, the only way this is called is inside of the package, and the callsite is from the handler themselves, so we should not reach such a case.
I think if that happens, it might as well be a crash that we should investigate and fix ASAP. We could replace the nil-ref panic by a more explicit one, but at that point I'm not sure it'd help too much, what do you think?